python/pyspark/sql/column.py

```diff
@@ -71,6 +71,18 @@ def _to_java_column(col: "ColumnOrName") -> JavaObject:
     return jcol
 
 
+def _to_java_expr(col: "ColumnOrName") -> JavaObject:
+    if isinstance(col, (Column, str)):
```

Comment:
I am saying that there's a duplicate if-else here so the else branch here is unreachable.